### PR TITLE
#174838693 Implement sign up with Google feature

### PR DESCRIPTION
**Pull request by AbonyiXavier**
**What does this PR do**
- This PR enables User to  signup and sign in with google OAuth feature

**Description of tasks to be completed?**
- Install relevant dependencies using `npm install`
- Write a program to  signup and sign in a user using google oauth
- Test program to ensure the user is signed in successfully

**How should this be manually tested/checked?**
- Clone the PR
- Cd into the branch ft-user-sign-in-174566162
- Install the dependencies using `npm install`
- Start the server by running `npm run dev`
- Create a .env file and define the following as follows:
  DEV_DATABASE_URL = `<development database url>`
  TEST_DATABASE_URL = `<test database url>`
  DATABASE_URL = `<production database url>`
- For testing purpose only and will be removed very soon
  GOOGLE_CLIENT_ID = `963856301923-dng2v1l2dla6qnn80pv7np9qcol8l2cg.apps.googleusercontent.com`
  GOOGLE_CLIENT_SECRET = ` CzpaH1Hj5Dc-MUyrboJdZZex`
  GOOGLE_CALLBACK_URL = `http://localhost:3000/auth/google/callback`
  COOKIE_KEY = `cookieKey`
- To test this feature go to  `http://localhost:3000/api/v1/auth/google` as URL to signup.
- User details get shown up on the browser letting you know its working for now

**What are the relevant pivotal tracker stories?**
[#174838693](https://www.pivotaltracker.com/story/show/174838693)

### CODE REVIEW

File: src/database/config/passport.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { Strategy } from "passport-google-oauth20";
```
Comment (codeliezel) [inline src/database/config/passport.js:1]: Now for this config, you don't need to add it to database -> config again, just add it to config-> config. that is where it should be. they are not db files.

File: src/models/user.js [hunk 2]
```diff
@@ -13,10 +13,10 @@ export default (sequelize, DataTypes) => {
       type: DataTypes.STRING,
       allowNull: false,
     },
-    role: {
-      type: DataTypes.ENUM("Super Admin", "Admin", "User"),
-      defaultValue: "user",
-    },
+    // role: {
```
Comment (codeliezel) [inline src/models/user.js:16]: this is wrong, please don't send commented out code to a codebase again.

**Review by codeliezel — CHANGES_REQUESTED** [review]
Why have you not rebase, also before pushing your code, lint them very very well? Why is code climate also shouting.

**Review by codeliezel — CHANGES_REQUESTED** [review]
Now, maybe you don't get the requirements for this PR. 

- the OAuth is not signing me up, it's meant to if I am a first time user
- then if it's that I'm signed up already, and I try signing up with my google, it should bring out an error that user exists.
- then as a user is signing up, I want to see the data in the DB. The data has to be filled in the users table to confirm if a user has signed up.

**Comment by codeliezel** [comment]
when I use this link again: `http://localhost:3000/auth/google/callback`
- I get this error
dev1

**Review by bellogo — APPROVED** [review]

**Review by balldash — DISMISSED** [review]

**Review by donaldcrane — DISMISSED** [review]

**Review by bellogo — APPROVED** [review]

**Review by codeliezel — APPROVED** [review]

**Pull request merged**